src/textual/strip.py

```diff
@@ -109,8 +109,8 @@ def __reversed__(self) -> Iterator[Segment]:
     def __len__(self) -> int:
         return len(self._segments)
 
-    def __eq__(self, strip: Strip) -> bool:
-        return (
+    def __eq__(self, strip: object) -> bool:
```

Comment:
This broke LSP on `__eq__`, which unlike __gt__ etc must be able to receive an object, not just the generic type param. I changed the logic (not really change, is it?) below for robustness and type safety accordingly.